chat-plugins/uno.js

```diff
@@ -0,0 +1,547 @@
+	}
+}
+
+exports.commands = {
+	uno: {
+		// roomowner commands
+		disable: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (room.unoDisabled) {
+				return this.errorReply("UNO is already disabled.");
+			}
+			room.unoDisabled = true;
+			if (room.chatRoomData) {
+				room.chatRoomData.unoDisabled = true;
+				Rooms.global.writeChatRoomData();
+			}
+			return this.sendReply("UNO has been disabled for this room.");
+		},
+
+		enable: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (!room.unoDisabled) {
+				return this.errorReply("UNO is already enabled.");
+			}
+			delete room.unoDisabled;
+			if (room.chatRoomData) {
+				delete room.chatRoomData.unoDisabled;
+				Rooms.global.writeChatRoomData();
+			}
+			return this.sendReply("UNO has been enabled for this room.");
+		},
+
+		// moderation commands
+		create: function (target, room, user) {
+			if (!this.can("minigame", null, room)) return;
+			if (room.unoDisabled) return this.errorReply("UNO is disabled for this room.");
+			if (room.game) return this.errorReply("There is already a game in progress in this room.");
+
+			room.game = new UNOgame(room, target);
+			room.add(`|uhtml|uno-${room.gameNumber}|<div class="broadcast-green"><p style="font-size: 14pt; text-align: center;">A new game of <strong>UNO</strong> is starting!</p><p style="font-size: 9pt; text-align: center;">Use <strong>/uno join</strong> to join the game.</p></div>`).update();
+			this.privateModCommand(`(A game of UNO was created by ${user.name}.)`);
+		},
+
+		start: function (target, room, user) {
+			if (!this.can("minigame", null, room)) return;
+			if (!room.game || room.game.gameid !== "uno" || room.game.state !== "signups") return this.errorReply("There is no UNO game in signups phase in this room.");
+			if (room.game.onStart()) this.privateModCommand(`(The game of UNO was started by ${user.name}.)`);
+		},
+
+		end: function (target, room, user) {
+			if (!this.can("minigame", null, room)) return;
+			if (!room.game || room.game.gameid !== "uno") return this.errorReply("There is no UNO game going on in this room.");
+			room.game.destroy();
+			room.add("The game of UNO was forcibly ended.").update();
+			this.privateModCommand(`(The game of UNO was ended by ${user.name}.)`);
+		},
+
+		timer: function (target, room, user) {
+			if (!this.can("minigame", null, room)) return;
+			if (!room.game || room.game.gameid !== "uno") return this.errorReply("There is no UNO game going on in this room.");
+			let amount = parseInt(target);
+			if (!amount || amount < 5) return this.errorReply("The amount must be a number greater than 5.");
+
+			room.game.maxTime = amount;
+			if (room.game.timer) {
+				clearTimeout(room.game.timer);
+				room.game.timer = setTimeout(() => {
+					room.game.eliminate(room.game.currentPlayer);
+				}, amount * 1000);
```

Comment:
There needs to be a cap though. The timeout should be less than 2^31 so that it fits in a 32-bit integer rather than overflowing.

Reject the input if ``amount * 1000 > 0x7FFFFFFF``